JustSaying.Sample.Restaurant.KitchenConsole/Program.cs

```diff
@@ -0,0 +1,55 @@
+using Amazon;
+using JustSaying.Sample.Restaurant.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace JustSaying.Sample.Restaurant.KitchenConsole
+{
+    internal class Program
+    {
+        private static async Task Main()
+        {
+            Console.Title = "KitchenConsole";
+
+            await new HostBuilder()
+               .ConfigureLogging((hostContext, logging) =>
+               {
+                   logging.AddConsole();
+               })
+              .ConfigureServices((hostContext, services) =>
+              {
+                  services.AddJustSaying(config =>
+                  {
+                      config.Client(x =>
+                      {
+                          // Use the simulated pafortin/goaws docker image
+                          x.WithServiceUrl("http://localhost:4100");
+
+                          // or supply credentials if using the real aws services
+                          //x.WithCredentials(...)
+                      });
+                      config.Messaging(x =>
+                      {
+                          x.WithRegion(RegionEndpoint.APSoutheast2);
+                      });
+                      config.Subscriptions(x =>
+                      {
+                          x.ForTopic<OrderPlacedEvent>();
+                      });
+                      config.Publications(x =>
+                      {
+                          x.WithTopic<OrderReadyEvent>();
+                      });
+                  });
+                  services.AddJustSayingHandler<OrderPlacedEvent, OrderPlacedEventHandler>();
+                  services.AddHostedService<Subscriber>();
```

Comment:
I think that's right - in fact I've had the idea of creating a built-in hosted service for JustSaying (that would be a new NuGet package) that would do some of this boiler plate to make it even easier to do something like this. We have a lot of worker services we use to process messages, and as we move more of them to .NET Core over time, it would be great to reduce as much repetitive boiler plate code for such use cases as possible.

For example `services.AddJustSayingService<MyService>()` where there could be a base class that could be derived from to do configuration, and then everything else gets bootstrapped up.

Of course that's a totally separate thing, but if we get this nicely working in the sample it would be a good jumping off point to investigate doing something like that.

Reply:
One thing i wanted to explore further was building a cron scheduled background task then integrating that with JustSaying so that I could publish a message on a schedule. My thinking was it would help decouple what a service does from how and when it is triggered
`*/10 * * * *` -> `OrderExpiryCheckRequiredEvent`
`0 */4 * * *` -> `OrderCancellationSummaryRequiredEvent`